ocaml/xapi/xapi_mgmt_iface.ml

```diff
@@ -160,31 +160,33 @@ let rebind ~__context =
 let ip_mutex = Mutex.create ()
 let ip_cond = Condition.create ()
 
-let wait_for_management_ip ~__context =
-  let ip = ref (match Helpers.get_management_ip_addr ~__context with Some x -> x | None -> "") in
-  let is_connected = ref (Helpers.get_management_iface_is_connected ~__context) in
-  Mutex.execute ip_mutex
-    (fun () -> while !ip = "" && !is_connected = false do
-          Condition.wait ip_cond ip_mutex;
-          ip := (match Helpers.get_management_ip_addr ~__context with Some x -> x | None -> "");
-          is_connected := (Helpers.get_management_iface_is_connected ~__context)
-        done);
-  !ip
+let wait_for_ip get_ip is_connected =
+  let rec loop () =
+    let ip = match get_ip () with Some x -> x | None -> "" in
+    let connected = is_connected () in
+    if ip = "" || not connected then begin
```

Comment:
I could've just compared against None here, but we need the string for the debug line below, and in case something goes wrong and we get `Some ""` for the IP, we catch that here (although by looking at the code I don't think there is a way that could happen).